evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -396,6 +396,25 @@ pub fn apply<DB: DatabaseExt>(
                 },
             )??
         }
+        HEVMCalls::SetNonceUnsafe(inner) => with_journaled_account(
+            &mut data.journaled_state,
+            data.db,
+            inner.0,
+            |account| -> Result {
+                let new = inner.1;
+                account.info.nonce = new;
+                Ok(Bytes::new())
+            },
+        )??,
+        HEVMCalls::ResetNonce(inner) => with_journaled_account(
+            &mut data.journaled_state,
+            data.db,
+            inner.0,
+            |account| -> Result {
+                account.info.nonce = 0;
```

Comment:
One thing I just thought of is that contract nonces start at 1, so we probably need to check if the account has code and then set it to 1 or 0 based on that: https://eips.ethereum.org/EIPS/eip-161

Reply:
Pulled into https://github.com/foundry-rs/foundry/issues/5043 to track